src/api/itemGeolocation.ts

```diff
@@ -0,0 +1,59 @@
+import { Item, ItemGeolocation, UUID } from '@graasp/sdk';
+
+import { PartialQueryConfigForApi } from '../types';
+import { verifyAuthentication } from './axios';
+import {
+  buildDeleteItemGeolocationRoute,
+  buildGetItemGeolocationRoute,
+  buildGetItemsInMapRoute,
+  buildPutItemGeolocationRoute,
+} from './routes';
+
+// eslint-disable-next-line import/prefer-default-export
+export const getItemGeolocation = async (
+  { API_HOST, axios }: PartialQueryConfigForApi,
+  id: UUID,
+) =>
+  axios
+    .get<ItemGeolocation>(`${API_HOST}/${buildGetItemGeolocationRoute(id)}`)
+    .then(({ data }) => data);
+
+export const putItemGeolocation = async (
+  payload: { itemId: Item['id'] } & Pick<ItemGeolocation, 'lat' | 'lng'>,
+  { API_HOST, axios }: PartialQueryConfigForApi,
+) =>
+  verifyAuthentication(() =>
+    axios
+      .put<void>(
+        `${API_HOST}/${buildPutItemGeolocationRoute(payload.itemId)}`,
+        { lat: payload.lat, lng: payload.lng },
+      )
+      .then(({ data }) => data),
+  );
+
+export const getItemsInMap = async (
+  payload: {
+    lat1: ItemGeolocation['lat'];
+    lat2: ItemGeolocation['lat'];
+    lng1: ItemGeolocation['lng'];
+    lng2: ItemGeolocation['lng'];
```

Comment:
Do `1` and `2` define any special order ? or do you take whichever is the lowest and highest in the backend ? If they express an order would `latStart` and `latEnd` be more meaningful names ?

Reply:
the backend still reorder in case they were not correctly set, what do you think of the naming then?